Add various EntityTag tags/mecs

MusicScore opened the pull request:
Adds multiple 1.14-specific tags/mecs for entities. Specifically:
- cat types
- cat/wolf collar color
- vex charging state
- fox crouching
- fox type
- mooshroom variants
- panda genes
- pillager patrol leader status
- pillager patrol location target

Fixes #7 (vex-related issue)

MusicScore committed:
Add various tags/mecs

MusicScore committed:
Fix EntityPandaGene

I accidentally made it impossible for the property to ever be used, and I also misnamed a tag.

ghost commented:
I pushed a fix for EntityPandaGene. I may or may not have had a copy-paste oopsies moment there

MusicScore committed:
Additional fix for EntityPandaGene

I had forgotten to check whether the specified inputs were actually valid genes. This commit rectifies that.

ghost commented:
Pushed an additional fix that prevents users from using invalid gene inputs in the `genes` mechanism for pandas.

MusicScore committed:
Fix version enums for 1.13- and 1.14-specific properties, remove pointless conditional checks in getPropertyString() method

In hindsight, dEntity/EntityTag objects almost never return a set of properties in the form of e@ENTITY[property=value;...] when you try to get the entity via a tag anyways, so most of the conditional checks that I did for those methods were rather unnecessary.

ghost commented:
Force-pushed a fix for incompatibilities with the latest commits, plus some code cleanup.

EDIT: Additionally, tested to work on 1.12.2 to 1.14.4.
